app/src/main/java/com/duckduckgo/app/fire/DatabaseCleaner.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.fire
+
+import android.database.sqlite.SQLiteDatabase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+
+interface DatabaseCleaner {
+    suspend fun cleanDatabase(): Boolean
+}
+
+class AuthDatabaseCleaner(private val databaseLocator: DatabaseLocator) : DatabaseCleaner {
```

Comment:
I would suggest to rather pass directly the db path as oppose to passing the locator.
It took me a while to track down what DB was being cleaned-up. Given that we're passing an interface (`DatabaseLocator`) that has two implementations.

In the `BrowserModule` (and its stub) we could do

```kotlin
    @Provides
    fun authDatabaseCleaner(@Named("authDbLocator") authDatabaseLocator: DatabaseLocator): DatabaseCleaner = AuthDatabaseCleaner(authDatabaseLocator.getDatabasePath())
```